mitosheet/src/mito/utils/actions.tsx

```diff
@@ -1649,6 +1649,109 @@ export const getActions = (
             searchTerms: ['docs', 'documentation', 'help', 'support'],
             tooltip: "Documentation, tutorials, and how-tos on all functionality in Mito."
         },
+        [ActionEnum.Select_Columns]: {
+            type: 'build-time',
+            staticType: ActionEnum.Select_Columns,
+            longTitle: 'Select column',
+            actionFunction: () => {
+                // We turn off editing mode, if it is on
+                setEditorState(undefined);
+
+                // We close the editing taskpane if its open
+                closeOpenEditingPopups();
+
+                // We open the column selection taskpane
+                let newStartingColumnIndex = gridState.selections[0].startingColumnIndex;
+                let newEndingColumnIndex = gridState.selections[0].endingColumnIndex;
+                if (newStartingColumnIndex === -1) {
+                    newStartingColumnIndex = 0;
+                }
+                if (newEndingColumnIndex === -1) {
+                    newEndingColumnIndex = sheetData.numColumns - 1;
+                }
+                setGridState(prevGridState => {
+                    return {
+                        ...prevGridState,
+                        selections: [{
+                            startingColumnIndex: newStartingColumnIndex ?? 0,
+                            endingColumnIndex: newEndingColumnIndex ?? 0,
+                            startingRowIndex: -1,
+                            endingRowIndex: -1,
+                            sheetIndex: sheetIndex
+                        }]
+                    }
+                });
+            },
+            isDisabled: () => {return doesAnySheetExist(sheetDataArray) ? defaultActionDisabledMessage : 'There are no columns to select. Import data.'},
+            searchTerms: ['select', 'columns', 'select columns'],
+            tooltip: "Select columns for all cells in currently selected range."
+        },
+        [ActionEnum.Select_Rows]: {
+            type: 'build-time',
+            staticType: ActionEnum.Select_Rows,
+            longTitle: 'Select row',
+            actionFunction: () => {
+                // We turn off editing mode, if it is on
+                setEditorState(undefined);
+
+                // We close the editing taskpane if its open
+                closeOpenEditingPopups();
+
+                // We open the column selection taskpane
+                let newStartingRowIndex = gridState.selections[0].startingRowIndex;
+                let newEndingRowIndex = gridState.selections[0].endingRowIndex;
+                if (newStartingRowIndex === -1) {
+                    newStartingRowIndex = 0;
+                }
+                if (newEndingRowIndex === -1) {
+                    newEndingRowIndex = sheetData.numRows - 1;
```

Comment:
I don't think these checks make a ton of sense to me. The main thing here is that starting index and ending index can be out of order. E.g. it's totally valid to have a startingRowIndex=5 and an endingRowIndex=-1, which means the user just first clicked row 5, and then selected upward to a column header. 

In this example case, it feels like this would end up selecting from row 5 to the end of the sheet -- which is not what we want. 

It feels like what we wanna do is:
1. Find the minRowIndex = min(startingRowIndex, endingRowIndex)
2. Find the maxRowIndex = ...
3. Set newStartingRowIndex = max(minRowIndex, 0)
4. Set newEndingRowIndex = min(maxRowIndex, sheetData.numRows - 1)

Going to check and verify this bug is real -- but I think the logic in this action (and the other one) is a bit weird.

Reply:
There is currently a bug if you have multiple rows selected and do Shift+" " -- it only takes the first or last row.

See here: 

https://github.com/mito-ds/mito/assets/21002431/0c478dfb-0d9d-436d-b8a8-d7718f94d793